scripts/wallet.js

```diff
@@ -690,20 +690,42 @@ export class Wallet {
         return histTXs;
     }
     sync = lockableFunction(async () => {
+        if (!this.isLoaded()) {
+            throw new Error('Attempting to sync without a wallet loaded');
+        }
         if (this.#isSynced) {
             throw new Error('Attempting to sync when already synced');
         }
-        // While syncing the wallet ( DB read + network sync) disable the event balance-update
+        // While syncing the wallet (DB read + network sync) disable the event balance-update
         // This is done to avoid a huge spam of event.
         getEventEmitter().disableEvent('balance-update');
 
         await this.loadFromDisk();
         await this.loadShieldFromDisk();
-        // Let's set the last processed block 5 blocks behind the actual chain tip
-        // This is just to be sure since blockbook (as we know)
-        // usually does not return txs of the actual last block.
-        this.#lastProcessedBlock = blockCount - 5;
-        await this.#transparentSync();
+        this.#lastProcessedBlock = blockCount;
```

Comment:
What's the issue? Since `getblockcount` is from nodes now, `getLatestBlocks` (`getblock` RPC) is guaranteed to have the blocks up to `blockCount`, the delay was only necessary because of Blockbook's slow block indexing.

Reply:
if you have a `tx` at the exact current blockchain height and you are syncing, then `getLatestTx` will likely not return it. Therefore, if we set `lastProcessedBlock = blockCount`, that `tx` will never be received when calling `getLatestBlocks`